src/coreclr/tools/superpmi/superpmi/commandline.cpp

```diff
@@ -669,12 +659,170 @@ bool CommandLine::Parse(int argc, char* argv[], /* OUT */ Options* o)
+                break;
+            case SPMI_TARGET_ARCHITECTURE_X86:
+                targetArch = "x86";
+                break;
+            case SPMI_TARGET_ARCHITECTURE_ARM:
+                targetArch = "arm";
+                break;
+            case SPMI_TARGET_ARCHITECTURE_ARM64:
+                targetArch = "arm64";
+                break;
+            default:
+                allowDefaultJIT = false;
+                break;
+        }
+
+        size_t programPathLen = 0;
+        char* programPath = nullptr;
+        const char* lastSlash = strrchr(argv[0], DIRECTORY_SEPARATOR_CHAR_A);
+        if (lastSlash == nullptr)
+        {
+            allowDefaultJIT = false;
+        }
+        else
+        {
+            programPathLen = lastSlash - argv[0] + 1;
+            programPath = new char[programPathLen];
+            strncpy_s(programPath, programPathLen, argv[0], programPathLen - 1);
+        }
+
+        if (allowDefaultJIT)
+        {
+            const char* jitOSName = nullptr;
+
+            if (defaultSpmiTargetArchitecture != GetSpmiTargetArchitecture())
+            {
+                // SuperPMI doesn't know about "target OS" so always assume host OS.
+
+                switch (GetSpmiTargetArchitecture())
+                {
+                    case SPMI_TARGET_ARCHITECTURE_AMD64:
+                    case SPMI_TARGET_ARCHITECTURE_X86:
+                        jitOSName = hostOSTag;
+                        break;
+                    case SPMI_TARGET_ARCHITECTURE_ARM:
+                    case SPMI_TARGET_ARCHITECTURE_ARM64:
+                        jitOSName = "universal";
+                        break;
+                    default:
+                        // Can't get here if `allowDefaultJIT` was properly set above.
+                        break;
+                }
+            }
+
+            const char* const jitBaseName = "clrjit";
+            size_t len = programPathLen + strlen(jitHostOSPrefix) + strlen(jitBaseName) + strlen(jitHostOSExtension) + 1;
+            if (jitOSName != nullptr)
+            {
+                len += 3 /* underscores */ + strlen(jitOSName) + strlen(targetArch) + strlen(hostArch);
+            }
+            char* tempStr = new char[len];
+            if (jitOSName == nullptr)
+            {
+                sprintf_s(tempStr, len, "%s%c%s%s%s",
+                    programPath,
+                    DIRECTORY_SEPARATOR_CHAR_A,
+                    jitHostOSPrefix,
+                    jitBaseName,
+                    jitHostOSExtension);
+            }
+            else
+            {
+                sprintf_s(tempStr, len, "%s%c%s%s_%s_%s_%s%s",
+                    programPath,
+                    DIRECTORY_SEPARATOR_CHAR_A,
+                    jitHostOSPrefix,
+                    jitBaseName,
+                    jitOSName,
+                    targetArch,
+                    hostArch,
+                    jitHostOSExtension);
+            }
```

Comment:
Is there any reason to not just use the "default" JIT (clrjit.dll/libclrjit.so/libclrjit.dylib)?

Reply:
We actually do use the default. Only if you use the `-target` option do the non-default ones get used.